specs-jetpack-calypso/wp-jetpack-connect-spec.js

```diff
@@ -365,4 +372,70 @@ test.describe( `Jetpack Connect: (${ screenSize })`, function() {
 			assert( plansPage.confirmCurrentPlan( 'free' ) );
 		} );
 	} );
+
+	test.describe( 'Connect from Jetpack.com Pricing page and buy paid plan: @parallel @jetpack', function() {
+		this.bailSuite( true );
+
+		test.before( function() {
+			return driverManager.ensureNotLoggedIn( driver );
+		} );
+
+		test.it( 'We can set the sandbox cookie for payments', function() {
+			return ( new WPHomePage( driver, { visit: true, culture: locale } ).setSandboxModeForPayments( sandboxCookieValue ) );
+		} );
+
+		test.it( 'Can create wporg site', function() {
+			this.jnFlow = new JetpackConnectFlow( driver, null, 'noJetpack' );
+			return this.jnFlow.createJNSite();
+		} );
+
+		test.it( 'Can select buy Premium on Pricing Page', function() {
+			const jetpackComPage = new JetpackComPricingPage( driver );
+			return jetpackComPage.buyPremium();
+		} );
+
+		test.it( 'Can start connection flow using JN site', function() {
+			const connectPage = new JetpackConnectPage( driver );
+			return connectPage.addSiteUrl( this.jnFlow.url );
+		} );
+
+		test.it( 'Can click Install Jetpack button in the instructions page', function() {
+			const jetpackConnectInstall = new JetpackConnectInstallPage( driver, false );
+			return jetpackConnectInstall.clickInstallButton();
+		} );
+
+		test.it( 'Can click the install button in the wp-admin plugin iframe', function() {
+			const wpAdminPluginPopup = new WPAdminPluginPopup( driver );
+			return wpAdminPluginPopup.installPlugin();
+		} );
+
+		test.it( 'Can click the plugin Activate button in the wp-admin updates page', function() {
+			const wpAdminUpdatesPage = new WPAdminUpdatesPage( driver );
+			return wpAdminUpdatesPage.activatePlugin();
+		} );
+
+		test.it( 'Can click the Connect Jetpack button', function() {
+			const wpAdminPluginsPage = new WPAdminPluginsPage( driver );
+			return wpAdminPluginsPage.connectJetpackAfterActivation();
+		} );
+
+		test.it( 'Can log into WP.com', function() {
+			const user = dataHelper.getAccountConfig( 'jetpackConnectUser' );
+			const loginPage = new LoginPage( driver );
+			return loginPage.login( user[0], user[1] );
+		} );
+
+		test.it( 'Can see the secure payment page and enter/submit test payment details', function() {
```

Comment:
This failed a few times for me locally, because it was still on the previous "connecting" page which isn't mentioned in this flow, even though this connecting page disappears automatically it might be worth putting in a step that waits for it to disappear.